tasks.py

```diff
@@ -680,11 +680,53 @@ def hadolint(context):
     run_command(context, command)
 
 
-@task
-def pylint(context):
+@task(
+    help={
+        "ref": "`pylint-nautobot` git reference to use for the analysis, can be a local path to the repository"
+        ", or a git reference e.g. `--ref=develop`. Use currently installed version if empty. (default: Empty)",
+    }
+)
+def pylint(context, ref=""):
     """Run pylint code analysis."""
-    command = 'pylint --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml nautobot_dev_example'
-    run_command(context, command)
+    pylint_command = [
+        "pylint",
+        '--init-hook="import nautobot; nautobot.setup()"',
+        "--rcfile=pyproject.toml",
+        "nautobot_dev_example",
+    ]
+
+    if not ref:
+        run_command(context, " ".join(pylint_command))
+        return
+
+    pip_and_pylint_command = ["pip install"]
```

Comment:
Why do we have to `pip install` it here?

Reply:
For our implementation, similar to `pylint-django`, it's necessary to have the package installed.

Reply:
What I don't understand is why is this not installed by poetry?

Reply:
`poetry install` doesn't allow to specify the GitHub as a package source, works with local `pyproject.toml` only.